src/main/java/com/github/maven_nar/AbstractNarLayout.java

```diff
@@ -133,10 +133,14 @@ protected void unpackNarAndProcess(final ArchiverManager archiverManager, final
       NarUtil.makeLink(new File(narLocation, "lib/" + defaultAOL), this.log);
     }
     if (linkerName.equals(gcc) || linkerName.equals(gpp)) {
-      NarUtil.runRanlib(new File(narLocation, "lib/" + defaultAOL), this.log);
+      if (!skipRanlib) {
+        NarUtil.runRanlib(new File(narLocation, "lib/" + defaultAOL), this.log);
+  	  }
       // FIXME clumsy
       if (defaultAOL.hasLinker(gpp)) {
-        NarUtil.runRanlib(new File(narLocation, "lib/" + NarUtil.replace(gpp, gcc, defaultAOL.toString())), this.log);
+        if (!skipRanlib) {
```

Comment:
Wonder if it's style, but why not just surround the whole rather than test !skipRanlib twice?